src/Umbraco.Core/StringExtensions.cs

```diff
@@ -1470,5 +1470,29 @@ internal static Guid ToGuid(this string text)
             byte[] hash = md5.ComputeHash(myStringBytes);
             return new Guid(hash);
         }
+
+
+        /// <summary>
+        /// Converts a file name to a friendly name for a content item
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        public static string friendlyNameFromFilename(this string fileName)
+        {
+            // strip the file extension
+            fileName = StripFileExtension(fileName);
+
+            // underscores and dashes to spaces
+            fileName = ReplaceMany(fileName, new char[] { '_', '-' }, ' ');
+
+            // any other conversions ?
+
+            // Pascalcase (to be done last)
+            fileName = System.Threading.Thread.CurrentThread.CurrentCulture.TextInfo.ToTitleCase(fileName);

```

Comment:
We can't use `System.Threading.Thread.CurrentThread.CurrentCulture` because current culture changes based on who logs in which means we can't expect the exact same result of ToTitleCase between users. 

We actually have built in utilities to do casing which is part of `IShortStringHelper` and in fact we already have a method for this called `CleanStringForSafeFileName` which should be used.

I'll update this implementation when I'm done the v7 -> v8 merge